src/main/java/seedu/address/model/parcel/DeliveryDate.java

```diff
@@ -40,32 +43,46 @@ public DeliveryDate(String deliveryDate) throws IllegalValueException {
         requireNonNull(deliveryDate);
         String trimmedDate = deliveryDate.trim();
         DateFormat df = new SimpleDateFormat("dd-MM-yyyy");
+        df.setLenient(false);
 
-        if (isValidDate(trimmedDate)) {
+        // Check if input is in a format we can understand
+        if (!isValidDateFormat(trimmedDate)) {
+            // Check if input is in a format that PrettyTime(NLP) can understand
+            if (!isValidPrettyTimeDate(trimmedDate)) {
+                throw new IllegalValueException(MESSAGE_DELIVERY_DATE_CONSTRAINTS);
+            } else { // NLP appears to understand the intention, so we accept the input
+                List<Date> dates = new PrettyTimeParser().parse(trimmedDate);
+                this.date = dates.get(0);
+            }
+        } else { // We understand the intention, so we accept the input
             try {
                 this.date = formatDate(trimmedDate);
-                this.value = df.format(this.date);
-            } catch (ParseException e) {
+            } catch (ParseException e) { // date is in correct format, but not a valid date.
                 throw new IllegalValueException(MESSAGE_DELIVERY_DATE_CONSTRAINTS);
             }
-        } else {
-            throw new IllegalValueException(MESSAGE_DELIVERY_DATE_CONSTRAINTS);
         }
 
+        // Format date correctly
+        this.value = df.format(this.date);
     }
 
     /**
      * Formats the input date according to the list VALID_STRING_FORMATS and returns it.
      */
-    private static Date formatDate(String inputDate) throws ParseException {
+    private Date formatDate(String inputDate) throws ParseException {
+
         for (String formatString : VALID_STRING_FORMATS) {
             try {
-                return new SimpleDateFormat(formatString).parse(inputDate);
+                DateFormat df = new SimpleDateFormat(formatString);
+                df.setLenient(false);
+                return df.parse(inputDate);
             } catch (ParseException e) {
-                e.printStackTrace();
+                logger.info("Failed to fit input delivery date in current format, trying next format...");
```

Comment:
Wait.. are you sure?  don't need write too many logs (quite bad if the logs too full), you logger below should inform already

Reply:
To be honest this is sort of a hackish because it is an intentional empty catch block. I'm iteratively try-ing until I get a returnable value. Otherwise, I'll throw the actual exception.